### PR TITLE
[ESP8266] [Documentation] Add note that machine.RTC is not fully sup…

### PR DESCRIPTION
**Pull request by oceanites**
For the ESP8266 port, some methods of `machine.RTC` are not supported.

Add a note in the ESP8266 documentation, that `RTC.now()`, `RTC.init()`, `RTC.deinit()` and `RTC.irq()` with a `handler` keyword is currently not supported. 

Supported are only `RTC.ALARM0`, `RTC.alarm()`, `RTC.alarm_left()`, `RTC.datetime()`, `RTC.irq()` and `RTC.memory()`:
```python
>>> from machine import rtc
>>> rtc = RTC()
>>> dir(rtc)
['__class__', 'ALARM0', 'alarm', 'alarm_left', 'datetime', 'irq', 'memory']
```

This is mentioned in #3220, #3710 and in  #2701.

### CODE REVIEW

**Comment by dpgeorge** [comment]
Thanks for the contribution.  Although there are already a few notes in the docs mentioning that certain methods may not be available on any given port, I guess it makes sense to explicitly list the missing ones here, because it's a point of continuing confusion.

File: docs/esp8266/quickref.rst [hunk 1]
```diff
@@ -235,7 +235,13 @@ and is accessed via the :ref:`machine.I2C <machine.I2C>` class::
 Real time clock (RTC)
 ---------------------
 
-See :ref:`machine.RTC <machine.RTC>` ::
+See :ref:`machine.RTC <machine.RTC>` 
+
+.. note:: Not all methods are implemented: `RTC.now()`, `RTC.irq(handler=*)`
```
Comment (dpgeorge) [inline docs/esp8266/quickref.rst]: I'd suggest to shift this block down to after the example, as the information is more "additional details" than "very important", and first-time users would want to see the examples first.

Comment (ketograph) [reply docs/esp8266/quickref.rst]: Seems reasonable, I changed it in a new commit.

**Comment by dpgeorge** [comment]
Thanks for updating, rebased and merged in 04e7aa056316d7b4400b508caefe26c6b75384f8

**Pull request closed without merging**